src/main/java/com/structures/helpers/Structure.java

```diff
@@ -0,0 +1,422 @@
+        blockList = template.blocks.toArray(blockList);
+
+        for (int i = 0; i < blockList.length; i++)
+        {
+            IBlockState finalState = blockList[i].blockState.withMirror(settings.getMirror()).withRotation(settings.getRotation());
+            BlockPos finalPos = Template.transformedBlockPos(settings, blockList[i].pos);
+            Template.BlockInfo finalInfo = new Template.BlockInfo(finalPos, finalState, blockList[i].tileentityData);
+            blockList[i] = finalInfo;
+        }
+        return blockList;
+    }
+
+    public Entity[] getEntityInfo(World world, BlockPos pos)
+    {
+        Template.EntityInfo[] entityInfoList = new Template.EntityInfo[template.entities.size()];
+        entityInfoList = template.blocks.toArray(entityInfoList);
+
+        Entity[] entityList = null;
+
+        for (int i = 0; i < entityInfoList.length; i++)
+        {
+            Entity finalEntity = EntityList.createEntityFromNBT(entityInfoList[i].entityData, world);
+            Vec3d entityVec = entityInfoList[i].pos.add(new Vec3d(pos));
+            finalEntity.setPosition(entityVec.xCoord, entityVec.yCoord, entityVec.zCoord);
+        }
+
+        return entityList;
+    }
+
+    public Entity[] getEntityInfoWithSettings(World world, BlockPos pos, PlacementSettings settings)
+    {
+        Template.EntityInfo[] entityInfoList = new Template.EntityInfo[template.entities.size()];
+        entityInfoList = template.entities.toArray(entityInfoList);
+
+        Entity[] entityList = new Entity[entityInfoList.length];
+
+        for (int i = 0; i < entityInfoList.length; i++)
+        {
+            Entity finalEntity = EntityList.createEntityFromNBT(entityInfoList[i].entityData, world);
+            Vec3d entityVec = Structure.transformedVec3d(settings, entityInfoList[i].pos).add(new Vec3d(pos));
+
+            if(finalEntity != null)
+            {
+                finalEntity.prevRotationYaw = (float) (finalEntity.getMirroredYaw(settings.getMirror()) - NINETY_DEGREE);
+                double rotation = (double) finalEntity.getMirroredYaw(settings.getMirror()) + ((double)finalEntity.rotationYaw - finalEntity.getRotatedYaw(settings.getRotation()));
+                finalEntity.setLocationAndAngles(entityVec.xCoord, entityVec.yCoord, entityVec.zCoord, (float) rotation, finalEntity.rotationPitch);
+            }
+            entityList[i] = finalEntity;
+        }
+
+        return entityList;
+    }
+
+    public BlockPos getSize(Rotation rotation)
+    {
+        return this.template.transformedSize(rotation);
+    }
+
+    public void setPlacementSettings(PlacementSettings settings)
+    {
+        this.settings = settings;
+    }
+
+    private static Vec3d transformedVec3d(PlacementSettings settings, Vec3d vec)
+    {
+        Mirror mirrorIn = settings.getMirror();
+        Rotation rotationIn = settings.getRotation();
+        double xCoord = vec.xCoord;
+        double yCoord = vec.yCoord;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'yCoord' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)